starter_code/src/movies.js

```diff
@@ -1,15 +1,2111 @@
+    genre: ['Drama', 'Romance'],
+    rate: '8.1'
+  },
+  {
+    title: 'Sholay',
+    year: '1975',
+    director: 'Ramesh Sippy',
+    duration: '3h 18min',
+    genre: ['Action', 'Adventure', 'Comedy', 'Drama', 'Musical', 'Thriller'],
+    rate: '8.2'
+  },
+  {
+    title: 'Beauty and the Beast',
+    year: '1991',
+    director: 'Gary Trousdale',
+    duration: '1h 24min',
+    genre: ['Animation', 'Family', 'Fantasy', 'Musical', 'Romance'],
+    rate: '8.0'
+  },
+  {
+    title: 'La battaglia di Algeri',
+    year: '1966',
+    director: 'Gillo Pontecorvo',
+    duration: '2h 1min',
+    genre: ['Drama', 'War'],
+    rate: '8.1'
+  },
+  {
+    title: 'Ah-ga-ssi',
+    year: '2016',
+    director: 'Chan-wook Park',
+    duration: '2h 24min',
+    genre: ['Crime', 'Drama', 'Mystery', 'Romance', 'Thriller'],
+    rate: '8.1'
+  },
+  {
+    title: 'Pirates of the Caribbean: The Curse of the Black Pearl',
+    year: '2003',
+    director: 'Gore Verbinski',
+    duration: '2h 23min',
+    genre: ['Action', 'Adventure', 'Fantasy'],
+    rate: '8.0'
+  },
+  {
+    title: 'PK',
+    year: '2014',
+    director: 'Rajkumar Hirani',
+    duration: '2h 33min',
+    genre: ['Comedy', 'Drama', 'Fantasy', 'Sci-Fi'],
+    rate: '8.2'
+  },
+  {
+    title: 'Dog Day Afternoon',
+    year: '1975',
+    director: 'Sidney Lumet',
+    duration: '2h 5min',
+    genre: ['Biography', 'Crime', 'Drama', 'Thriller'],
+    rate: '8.0'
+  },
+  {
+    title: 'Dead Poets Society',
+    year: '1989',
+    director: 'Peter Weir',
+    duration: '2h 8min',
+    genre: ['Comedy', 'Drama'],
+    rate: '8.0'
+  }
+];
 
-// Get the average of all rates with 2 decimals 
 
 
-// Get the average of Drama Movies
 
 
-// Order by time duration, in growing order
+// Get the average of all rates with 2 decimals
 
+//
+var totalRates = 0;
+function ratesAverage (array){
+  //This function sums the rates of all movies
+  totalRates = array.reduce(function (acc, aFilm){
+    return acc + parseFloat(aFilm.rate);
+  }, 0);
+
+  // an average is equal to the total amount of an array of numbers divided by the number of elements
+  var averageRate = totalRates/array.length;
+
+  //we should return the average rounded to two decimals, unless the average is an integer
+  if (Number.isInteger(averageRate)=== true){
+    return averageRate;
+  } else {
+    return parseFloat(averageRate.toFixed(2));
+  }
+}
+
+ratesAverage(movies);
+console.log(ratesAverage(movies));
+
+// ITERATION 2 : Get the average of Drama Movies
+
+//We create a new array composed of only drama movies
+var dramaMovies = movies.filter(function(aMovie){
+  return aMovie.genre.includes('Drama')
+});
+
+//we create a new array composed only of drama movies that have a rate property
+// var ratedDramaMovies = dramaMovies.filter(function(aDramaMovie){
+//   return aDramaMovie.rate;
+// });
+
+var ratedDramaMovies = [];
+var nnn=0
+
+dramaMovies.forEach(function (dramaMovie){
+  if (!dramaMovie.rate){
```

Comment:
not sure what you try to did with this if statement, but when you remove it, it will actually put some content in your array, otherwise your array will stay empty ;)

Reply:
Ok I understand what you've try to do, but you should do it like this : 

```js 

dramaMovies = [{ name: "hello", rate: 8 }, { name: "prout", rate: "" }];
dramaMovies.forEach(function(dramaMovie) {
  if (dramaMovie.rate == "") {
    dramaMovie.rate = 0;
    ratedDramaMovies.push(dramaMovie);
  } else ratedDramaMovies.push(dramaMovie);
});

```

We have the excpected output (4) but for some reason it's not working on jasmine.. I'm gonna take a look at it